tests/Functional/BackedEnumResourceTest.php

```diff
@@ -0,0 +1,200 @@
+            'links' => [
+                'self' => '/availabilities',
+            ],
+            'meta' => [
+                'totalItems' => 3,
+            ],
+            'data' => [
+                [
+                    'id' => '/availabilities/AVAILABLE',
+                    'type' => 'Availability',
+                    'attributes' => [
+                        'value' => 10,
+                    ],
+                ],
+                [
+                    'id' => '/availabilities/CANCELLED',
+                    'type' => 'Availability',
+                    'attributes' => [
+                        'value' => 20,
+                    ],
+                ],
+                [
+                    'id' => '/availabilities/POSTPONED',
+                    'type' => 'Availability',
+                    'attributes' => [
+                        'value' => 30,
+                    ],
+                ],
+            ],
+        ]);
+    }
+
+    public function testCollectionHal(): void
+    {
+        self::createClient()->request('GET', '/availabilities', ['headers' => ['Accept' => 'application/hal+json']]);
+
+        $this->assertResponseIsSuccessful();
+        $this->assertJsonEquals([
+            '_links' => [
+                'self' => [
+                    'href' => '/availabilities',
+                ],
+                'item' => [
+                    ['href' => '/availabilities/AVAILABLE'],
+                    ['href' => '/availabilities/CANCELLED'],
+                    ['href' => '/availabilities/POSTPONED'],
+                ],
+            ],
+            'totalItems' => 3,
+            '_embedded' => [
+                'item' => [
+                    [
+                        '_links' => [
+                            'self' => ['href' => '/availabilities/AVAILABLE'],
+                        ],
+                        'value' => 10,
+                    ],
+                    [
+                        '_links' => [
+                            'self' => ['href' => '/availabilities/CANCELLED'],
+                        ],
+                        'value' => 20,
+                    ],
+                    [
+                        '_links' => [
+                            'self' => ['href' => '/availabilities/POSTPONED'],
+                        ],
+                        'value' => 30,
+                    ],
+                ],
+            ],
+        ]);
+    }
+
+    public function testCollectionJsonLd(): void
+    {
+        self::createClient()->request('GET', '/availabilities', ['headers' => ['Accept' => 'application/ld+json']]);
+
+        $this->assertResponseIsSuccessful();
+        $this->assertJsonEquals([
+            '@context' => '/contexts/Availability',
+            '@id' => '/availabilities',
+            '@type' => 'hydra:Collection',
+            'hydra:totalItems' => 3,
+            'hydra:member' => [
+                [
+                    '@id' => '/availabilities/AVAILABLE',
```

Comment:
this is wrong on an RDF point of view, and as I pointed I have an ongoing work but can't merge/work on that right now